frontend/src/pages/DashboardHome.vue

```diff
@@ -49,13 +49,25 @@
                             </template>
 
                             <!-- Agent Display Name -->
-                            <span v-if="endpoint === ''">{{ $t("currentEndpoint") }}</span>
-                            <a v-else :href="agent.url" target="_blank">{{ endpoint }}</a>
+                            <template v-if="$root.agentStatusList[endpoint]">
+                                <span v-if="endpoint === '' && agent.friendlyname === ''" class="badge bg-secondary me-2">Controller</span>
+                                <span v-else-if="agent.friendlyname === ''" :href="agent.url">{{ endpoint }}</span>
+                                <span v-else :href="agent.url">{{ agent.friendlyname }}</span>
+                            </template>
```

Comment:
Couple of suggestions here: 

1. We can use the fact that an empty string is falsy to shorten this chain.
2. You probably don't want to be modifying things so drastically. If it's working, then a small change will do.

```suggestion
                            <span v-if="endpoint === ''" class="badge bg-secondary me-2">{{ $t("currentEndpoint") }}</span>
                            <a v-else :href="agent.url" target="_blank">{{ agent.name || endpoint }}</a>
```